accepted/2021/User-Experience/Apple-Silicon-UX.md

```diff
@@ -0,0 +1,107 @@
+
+#### Implications
+
+- The proposed version policy will prohibit us from making a native port of earlier runtime releases. .NET 2.1 will be out of support before .NET 6.0 releases. .NET 5.0 will be out of support 3 month after release of .NET 6. .NET 3.1 is the only viable candidate for back porting Apple Silicon support.  However, given the size of the changes to the runtime and the JIT it is highly unlikely this is something we would actively pursue.
+
+- The proposed version policy will prevent .NET 6 apps published with rid of osx-x64 from running on Apple Silicon even with Rosetta 2 emulation. This is desirable because long term we want to stop supporting Rosetta 2 emulation on Apple Silicon. IT support grows our effective platform support matrix and increases our support burden. We are dependent on Apple to fix Rosetta 2 issues, but they are dependent on us to root cause them. Performance is also expected to be better with native code.
+
+#### Future optimizations
+
+The current proposed design will perform the runtime search twice when an architecture mismatch is detected. The user can prevent this extra delay by calling dotnet with the correct architecture for the required runtime.
+
+- The default architecture is selected based on the architecture of the current process (typically the shell or the IDE).
+- The user or the IDE can explicitly select the architecture of the target process using the `arch` command.
+
+The extra search cost could be minimized by passing the result of the previous search when restarting the process. Adding an `--fx-version` option would accomplish this at the cost of changing the users specified argument order.  Using a new environment variable would avoid this unexpected side-effect.
+
+## Apple Silicon requires Signed Binaries
+
+Apple Silicon with Big Sur adds a new security policy which requires signed binaries by default. Unsigned binaries are simply killed without explanation. The user can expect an inscrutable message like `zsh killed: helloWorld` This directly affects our users who expect published binaries to just work.
+
+The developer would need to explicitly sign after publishing for Apple Silicon.  The sigining command is relatively simple `codesign -s <signature> <app-path>`.
+
+### Proposed changes
+
+This is largely TBD. The addition of a message during publish is the most likely design.
+
+## Universal Binaries
+
+As part of the architectural transition, Apple has switched the entire BigSur OS to use universal binaries. This means every file in the OS has both the `arm64` and the `x86_64` code on disk.
+
+Given both architectures, universal binaries are relatively easy to create. The general syntax would be `lipo <binaryArchA> <binaryArchB> -create -output <binaryUniversal>`
+
+### No planned universal binary .NET runtime
+
+At least for .NET 6, general universal binary support is not planned.
```

Comment:
I'd suggest rewording that. The universal binary support works already or is planned for Apple apps (bundles).